pkg/api/v1/project_teams.go

```diff
@@ -28,7 +28,7 @@ type Team struct {
 }
 
 func (in *Team) ToAtlas(teamID string) admin.TeamRole {
-	roleNames := make([]string, 0, len(in.Roles))
+	roleNames := make([]string, 0)
```

Comment:
```suggestion
	roleNames := make([]string, 0, len(in.Roles))
```

Reply:
@s-urbaniak I've been confused about this. If we pre-allocate a slice of some length and then we append onto this slice, we effectively have a slice which is twice the length we actually need, with the first half full of nil/"" (created at make([]string, 0, len(in,Roles) ) and the other half with the actual information. Is this not a correct assumption of the behaviour of make and append?

Reply:
I've investigated this and I understand now. This confusion came about when I had initialised a var as follows `mySlice := make([]string, len(otherSlice))`, resulting in a slice full of "" strings and all subsequent additions were appended to the end of this. In actuality, with the implementation you've provided, len(in.Roles) strings of memory is allocated to the var but the slice is of size 0. All appends to the slice (within this length) will be stored within this allocated memory. 👍